interface/ApfpInterfaceType.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+#include "Config.h"
+
+#ifdef APFP_GMP_INTERFACE_TYPE // Interface with GMP types
+#include <gmp.h>
+using ApfpInterfaceType = mpf_t;
+using ApfpInterfaceTypePtr = mpf_ptr;
+using ApfpInterfaceTypeConstPtr = mpf_srcptr;
+
+#else
+#include <mpfr.h>
+using ApfpInterfaceType = mpfr_t;
+using ApfpInterfaceTypePtr = mpfr_ptr;
+using ApfpInterfaceTypeConstPtr = mpfr_srcptr;
+#endif
+
+void InitApfpInterfaceType(ApfpInterfaceTypePtr value);
+
+void Init2ApfpInterfaceType(ApfpInterfaceTypePtr value, unsigned long precision);
+
+void ClearApfpInterfaceType(ApfpInterfaceTypePtr value);
+
+void SwapApfpInterfaceType(ApfpInterfaceTypePtr a, ApfpInterfaceTypePtr b);
+
+void SetApfpInterfaceType(ApfpInterfaceTypePtr dest, ApfpInterfaceTypeConstPtr source);
+
+void SetApfpInterfaceType(ApfpInterfaceTypePtr dest, long int source);
```

Comment:
When in a namespace, all of these could just be their GMP/MPFR name:

```suggestion
void set(ApfpInterfaceTypePtr dest, long int source);
```